chromium_src/ios/chrome/browser/ui/webui/about_ui.cc

```diff
@@ -0,0 +1,239 @@
+      if (!base::Contains(histogram_name, path)) {
+        continue;
+      }
+      base::Value::Dict histogram_dict = histogram->ToGraphDict();
+      std::string* header = histogram_dict.FindString("header");
+      std::string* body = histogram_dict.FindString("body");
+
+      response.append("<PRE>");
+      response.append("<h4>");
+      response.append(base::EscapeForHTML(*header));
+      response.append("</h4>");
+      response.append(base::EscapeForHTML(*body));
+      response.append("</PRE>");
+      response.append("<br><hr><br>");
+    }
+  }
+
+  FinishDataRequest(response, std::move(callback));
+}
+
+void AboutUIHTMLSource::FinishDataRequest(
+    const std::string& html,
+    web::URLDataSourceIOS::GotDataCallback callback) {
+  std::move(callback).Run(base::MakeRefCounted<base::RefCountedString>(html));
+}
+
+std::string AboutUIHTMLSource::GetMimeType(const std::string& path) const {
+  if (path == kCreditsJsPath || path == kStringsJsPath) {
+    return "application/javascript";
+  }
+
+  if (path == kCreditsCssPath) {
+    return "text/css";
+  }
+
+  return "text/html";
+}
+
+bool AboutUIHTMLSource::ShouldDenyXFrameOptions() const {
+  return web::URLDataSourceIOS::ShouldDenyXFrameOptions();
+}
+
+std::string AboutUIHTMLSource::ChromeURLs() const {
+  std::string chrome_urls = ::ChromeURLs();
+
+  // Replace Chrome -> Brave.
+  const std::string chrome_header = "Chrome URLs";
+  const std::string brave_header = "Brave URLs";
+  const std::string chrome_pages_header = "List of Chrome URLs";
+  const std::string brave_pages_header = "List of Brave URLs";
+  const std::string chrome_internal_pages_header =
+      "List of chrome://internals pages";
+  const std::string brave_internal_pages_header =
+      "List of brave://internals pages";
+  const std::string chrome_url_list = ">chrome://";
+  const std::string brave_url_list = ">brave://";
+  RE2::GlobalReplace(&chrome_urls, chrome_header, brave_header);
+  RE2::GlobalReplace(&chrome_urls, chrome_pages_header, brave_pages_header);
+  RE2::GlobalReplace(&chrome_urls, chrome_internal_pages_header,
+                     brave_internal_pages_header);
+  RE2::GlobalReplace(&chrome_urls, chrome_url_list, brave_url_list);
+
+  // Remove some URLs.
+  auto html_lines = base::SplitStringPiece(
+      chrome_urls, "\n", base::KEEP_WHITESPACE, base::SPLIT_WANT_ALL);
+  const base::flat_set<std::string_view> kURLsToRemove{
+      "brave://memories",
+  };
+  // URLs in html should be sorted so it's okay to iterate over sorted
+  // kURLsToRemove.
+  auto html_line_it = html_lines.begin();
+  auto url_to_remove_it = kURLsToRemove.begin();
+  while (html_line_it != html_lines.end() &&
+         url_to_remove_it != kURLsToRemove.end()) {
+    if (base::Contains(*html_line_it, *url_to_remove_it)) {
+      html_line_it = html_lines.erase(html_line_it);
+      ++url_to_remove_it;
+    } else {
+      ++html_line_it;
+    }
+  }
+
+  return base::JoinString(html_lines, "\n");
```

Comment:
looks like this is copied from `brave/chromium_src/chrome/browser/ui/webui/about_ui.cc`

please move this into `brave/components/webui` as a separate helper and share between ios/desktop (maybe with some flags if a special handling for iOS is required).

Reply:
What do I do about the special handling? Almost all of `WebUI` on iOS is different from on Desktop.
iOS uses `web::URLDataSourceIOS`.
Also iOS has to override the entire file since I can't override just the one function like on Desktop.